components/ai_chat/content/browser/page_content_fetcher.cc

```diff
@@ -348,11 +424,35 @@ std::string ExtractPdfContent(const ui::AXNode* pdf_root) {
   return pdf_content;
 }
 
+// Obtains a patch URL from a pull request URL
+std::optional<GURL> GetGithubPatchURLForPRURL(const GURL& url) {
+  if (!url.is_valid() || !url.SchemeIsCryptographic() ||
```

Comment:
would be better to check if scheme is exactly `https` unless we want to match other schemes.

Reply:
Fixed in https://github.com/brave/brave-core/pull/22289/commits/60cbe140539a6a15af691030a9828e2be89846d9